packages/xml-views-quick-fix/test/quick-fix-id-spec.ts

```diff
@@ -0,0 +1,171 @@
+    });
+
+    it("will get quick fix info when class is missing id attribute key - more than one id from same pattern", () => {
+      const testXmlSnippet = `<mvc:View
+                                xmlns:mvc="sap.ui.core.mvc"
+                                xmlns="sap.m"> 
+                                  <mvc:content>
+                                    <List id="_IDGenList2"></List>
+                                    <List id="_IDGenList1"></List>
+                                    <🢂List🢀$></List>
+                                  </mvc:content>
+                              </mvc:View>`;
+      const expectedSuggestion = ' id="_IDGenList3"';
+      const { document, start, end, idStartOffest } = getXmlSnippet(
+        testXmlSnippet
+      );
+      const testXmlDoc = getXmlDocument(document);
+      const quickFixInfo = getQuickFixIdInfo(testXmlDoc, { start, end });
+      expectExists(quickFixInfo, "Quick Fix Info");
+      expect(quickFixInfo.suggestion).to.equal(expectedSuggestion);
+      expect(quickFixInfo.offsetRange.start).to.equal(idStartOffest);
+    });
+
+    it("will get quick fix info when class has empty id attribute", () => {
+      const testXmlSnippet = `<mvc:View
+                                xmlns:mvc="sap.ui.core.mvc"
+                                xmlns="sap.m"> 
+                                <mvc:content>
+                                  <🢂List🢀 $id=""></List>
+                                </mvc:content>
+                              </mvc:View>`;
+      const expectedSuggestion = 'id="_IDGenList1"';
+      const { document, start, end, idStartOffest } = getXmlSnippet(
+        testXmlSnippet
+      );
+      const testXmlDoc = getXmlDocument(document);
+      const quickFixInfo = getQuickFixIdInfo(testXmlDoc, { start, end });
+      expectExists(quickFixInfo, "Quick Fix Info");
+      expect(quickFixInfo.suggestion).to.equal(expectedSuggestion);
+      expect(quickFixInfo.offsetRange.start).to.equal(idStartOffest);
+    });
+
+    it("will get quick fix info when class has empty id attribute between other attributes", () => {
+      const testXmlSnippet = `<mvc:View
+                                xmlns:mvc="sap.ui.core.mvc"
+                                xmlns="sap.m"> 
+                                <mvc:content>
+                                  <🢂List🢀 models="" $id="" footerText=""></List>
+                                </mvc:content>
+                              </mvc:View>`;
+      const expectedSuggestion = 'id="_IDGenList1"';
+      const { document, start, end, idStartOffest } = getXmlSnippet(
+        testXmlSnippet
+      );
+      const testXmlDoc = getXmlDocument(document);
+      const quickFixInfo = getQuickFixIdInfo(testXmlDoc, { start, end });
+      expectExists(quickFixInfo, "Quick Fix Info");
+      expect(quickFixInfo.suggestion).to.equal(expectedSuggestion);
+      expect(quickFixInfo.offsetRange.start).to.equal(idStartOffest);
+    });
+  });
```

Comment:
what happens if there are multiple attributes each in its own line?
would the insertion respect the existing indentation levels?

Reply:
in this case, the id will be generated in the tag line and other attributes will not be affected

Reply:
o.k, that is acceptable and may be upgraded in the future.

Lets add a couple of tests with multi line element:
- attributes exists on other lines.
- invalid id attribute exists on a different line than the opening tag